[en] Modify `load-balancer.md` (#1008)
cncf/glossary: content/en/load-balancer.md

## What it is

> A load balancer is a tool that efficiently distributes incoming requests among multiple instances of an application. Each service of a [microservices](/microservices/) app generally has multiple replicas (or instances), and the load balancer ensures no one service gets the bulk of the requests.

Reviewer: @sarkartanmay393, did you see my note in the issue? We already have this section. Let's build on what we have. You can take it as is or improve it but no need to start from scratch.

> A load balancer is a tool that distributes incoming requests among multiple instances of an application. Each service of a [microservices](/microservices/) app generally has multiple replicas (or instances), and the load balancer ensures no one service gets the bulk of the requests. Load balancers can be software or hardware-based.

> A load balancer is a tool that efficiently distributes incoming requests among multiple instances of an application. Each service of a [microservices](/microservices/) app generally has multiple replicas (or instances), and the load balancer ensures no one service gets the bulk of the requests.

Reviewer 2: Didn't recognize that there was already some proposal in the issue, because it is not linked. @CathPag @sarkartanmay393 I am ok with both either the original section mentioned in the issue or the suggestion I made. The only thing I do not agree is the statement that an app generally has multiple replicas/instances. It can have multiple instances, but not every app has multiple instances.

Load balancers can be software or hardware-based.

## Problem it addresses

Modern applications and websites serve thousands of simultaneous requests from end-users, and returning responses to each of that requests responsively, efficiently, and reliably is not easy.

> Previously network requests are routed to specific servers and in this high load scenario, some servers may get a very high number of requests and some servers stay comfortably.

Reviewer 2: ```suggestion
Unfortunately, [horizontal scaling](/horizontal-scaling/) introduces a new level of complexity which is, how to distribute the incoming traffic to all horizontally scaled services. This is where load balancers come in handy.
```

## How it helps

> Load balancer will distribute all the requests among multiple servers dynamically in a way where no one server get high load when there is other servers which can fulfill those requests. It evently spreads the load across multiple servers. Thus optimising every single request to get responed fast and reliably to end users.

Author: Hello @iamNoah1, I fixed some spelling mistakes and pushed that code again.

> A Load balancer will distribute all the requests among multiple servers dynamically in a way where no one server gets a high load when there are other servers that can fulfill those requests. It evenly spreads the load across multiple servers. Thus optimizing every single request to get responded fast and reliably to end-users.

Reviewer 2: ```suggestion
Load balancers will distribute all the requests among multiple servers dynamically in a way where no one server gets a high load when there are other servers that can fulfill those requests. It spreads the load across multiple servers, following a defined schema (i.e. evenly or percentage-based). Thus load balancers are playing an important role in the overall performance and user experience of an application.
```

> Load balancers will distribute all the requests among multiple servers dynamically in a way where no one server gets a high load when there are other servers that can fulfill those requests. It spreads the load across multiple servers, following a defined schema (i.e. evenly or percentage-based). Thus load balancers are playing an important role in the overall performance and user experience of an application.

Reviewer 2: ```suggestion
Load balancers will distribute all the requests among multiple servers dynamically in a way 
where no one server gets a high load when there are other servers that can fulfill those requests. 
It spreads the load across multiple servers, following a defined schema (i.e. evenly or percentage-based). 
Thus load balancers are playing an important role in the overall performance and user experience of an application.
```

## What it is

A load balancer is a tool that efficiently distributes incoming requests among multiple instances of an application. 
Take a [microservice](/microservices/) architecture for example, where each service can be [scaled horizontally](/horizontal-scaling/). 
A load balancer sits in front of a scaled microservice and ensures that no one instance gets the bulk of the requests.
Load balancers can be software or hardware-based.

## Problem it addresses

> Modern applications and websites serve thousands of thousands of simultaneous requests from end-users. 

Reviewer: ```suggestion
Modern applications and websites generally serve hundreds of thousands of simultaneous end-user requests. 
```

> In order to handle all those requests, [horizontal scaling](/horizontal-scaling/) is nearly unavoidable.

Reviewer: ```suggestion
To handle all those requests, applications are often [scaled horizontally](/horizontal-scaling/).
```

> Unfortunately, [horizontal scaling](/horizontal-scaling/) introduces a new level of complexity which is, how to distribute the incoming traffic to all horizontally scaled services. 

Reviewer: ```suggestion
But horizontal scaling introduces a new challenge. How do you distribute incoming traffic to all services equally? 
```

> This is where load balancers come in handy.

Reviewer: ```suggestion
This is where load balancers come in.
```

## How it helps

> Load balancers will distribute all the requests among multiple servers dynamically in a way 

Reviewer: ```suggestion
Load balancers dynamically distribute all incoming requests among multiple services, ensuring that no one service gets the bulk of it while others only get a few or none. 
```
Changed server to services since we are talking about modern (microservices-based) applications

> where no one server gets a high load when there are other servers that can fulfill those requests. 

Reviewer: ```suggestion

```
Part of the first sentence

> It spreads the load across multiple servers, following a defined schema (i.e. evenly or percentage-based). 

Reviewer: ```suggestion
In short, it spreads the load across multiple services, following a defined schema (i.e., evenly or percentage-based). 
```

> Thus load balancers are playing an important role in the overall performance and user experience of an application.

Reviewer: ```suggestion
Load balancers are essential to an application's overall performance and, ultimately, the user experience.
```
